presto-main/src/test/java/com/facebook/presto/sql/planner/TestTypeValidator.java

```diff
@@ -206,6 +209,33 @@ public void testValidAggregation()
         assertTypesValid(node);
     }
 
+    @Test
+    public void testValidIntermediateAggregation()
+    {
+        VariableReferenceExpression aggregationVariable = variableAllocator.newVariable("approx_percentile", VARBINARY);
+
+        PlanNode node = new AggregationNode(
+                Optional.empty(),
+                newId(),
+                baseTableScan,
+                ImmutableMap.of(aggregationVariable, new Aggregation(
+                        new CallExpression("approx_percentile",
+                                APPROX_PERCENTILE,
+                                VARBINARY,
+                                ImmutableList.of(variableC)),
+                        Optional.empty(),
+                        Optional.empty(),
+                        false,
+                        Optional.empty())),
+                singleGroupingSet(ImmutableList.of(variableA, variableB)),
```

Comment:
Intermediate aggregations are only be added for ungrouped aggregation, so would be better for the tests to also have empty grouping sets.  I don't know the reasoning for why it's only ungrouped aggregations (I think it's just not optimal rather than invalid, but if there is a reason it would invalid, we could add a plan validator for that too)

Reply:
In this case we don't really care (I think) about groups, but let me try making it empty.